### PR TITLE
Add compatibility mod for CSLA CDLC

### PR DESCRIPTION
**Pull request by Dahlgren**
**When merged this pull request will:**
- Add `ace_isBelt` for CSLA belt magazines
- Add backblast for CSLA launchers
- Add barrel swap to CSLA machine guns
- Add buddy Reloading for CSLA launchers
- Add open bolt to CSLA machine guns
- Add placeable CSLA US explosives

### TODO
- Add carrying to CSLA boxes
- Add dragging to CSLA boxes
- Add hearing protection to CSLA helmets
- Add gforce protection to CSLA pilot uniforms
- Add overheating to CSLA weapons
- Add placeable CSLA explosives
- Add rearm to CSLA vehicles
- Add refueling to CSLA vehicles
- Add repair to CSLA vehicles

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3mod.com/).
- [ ] [Development Guidelines](https://ace3mod.com/wiki/development/) are read, understood and applied.
- [ ] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by Drofseh** [comment]
Some weapons may need `ace_overheating_closedBolt` depending on where they inherit from.

https://github.com/acemod/ACE3/blob/master/addons/overheating/CfgWeapons.hpp

https://ace3mod.com/wiki/framework/overheating-framework.html#14-cook-off

**Comment by Dahlgren** [comment]
> Some weapons may need `ace_overheating_closedBolt` depending on where they inherit from.
> 
> https://github.com/acemod/ACE3/blob/master/addons/overheating/CfgWeapons.hpp
> 
> https://ace3mod.com/wiki/framework/overheating-framework.html#14-cook-off

Thanks! Added `ace_overheating_closedBolt = 0;` to M249, M60 and [UK59L](https://web.archive.org/web/20070621130009/http://world.guns.ru/machine/mg28-e.htm)

**Comment by Drofseh** [comment]
> > Some weapons may need `ace_overheating_closedBolt` depending on where they inherit from.
> > https://github.com/acemod/ACE3/blob/master/addons/overheating/CfgWeapons.hpp
> > https://ace3mod.com/wiki/framework/overheating-framework.html#14-cook-off
> 
> Thanks! Added `ace_overheating_closedBolt = 0;` to M249, M60 and [UK59L](https://web.archive.org/web/20070621130009/http://world.guns.ru/machine/mg28-e.htm)

You're welcome. Worth double checking your SMGs and sniper rifles as well, depending on where they inherit from.

**Comment by Dahlgren** [comment]
I'll move this to addons folder and add the `skipWhenMissingDependencies`

**Comment by PabstMirror** [comment]
we have no idea when 2.14 will drop
if these changes are good to go now we should think about merging as is and maybe just use the hasInclude
and then switch over to `skipWhen` after 2.14

**Comment by PabstMirror** [comment]
fixed
```
Updating base class CSLA_UK59_50rnd_7_62vz59->CSLA_Magazine, by z\ace\addons\compat_csla\config.cpp/CfgMagazines/CSLA_UK59_50rnd_7_62Sv59/ (original csla_cfg\config.bin)
Updating base class CSLA_UK59_50rnd_7_62vz59->CSLA_Magazine, by z\ace\addons\compat_csla\config.cpp/CfgMagazines/CSLA_UK59_50rnd_7_62PZ59/ (original csla_cfg\config.bin)
Updating base class CSLA_UK59_50rnd_7_62vz59->CSLA_Magazine, by z\ace\addons\compat_csla\config.cpp/CfgMagazines/CSLA_UK59_50rnd_7_62Tz59/ (original csla_cfg\config.bin)
Updating base class CSLA_UK59_50rnd_7_62vz59->CSLA_Magazine, by z\ace\addons\compat_csla\config.cpp/CfgMagazines/CSLA_UK59_50rnd_7_62Cv59/ (original csla_cfg\config.bin)
Updating base class US85_50Rnd_762x51->US85_30Rnd_762x51, by z\ace\addons\compat_csla\config.cpp/CfgMagazines/US85_200Rnd_762x51/ (original csla_us85_cfg\config.bin)
Updating base class CSLA_Launcher_AT->CSLA_LauncherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/CSLA_RPG7/ (original csla_cfg\config.bin)
Updating base class CSLA_Launcher_AT->CSLA_LauncherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/CSLA_RPG75/ (original csla_cfg\config.bin)
Updating base class US85_Launcher_AA->US85_launcherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/US85_FIM92/ (original csla_us85_cfg\config.bin)
Updating base class US85_Launcher_AT->US85_launcherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/US85_LAW72/ (original csla_us85_cfg\config.bin)
Updating base class US85_Launcher_AT->US85_launcherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/US85_M136/ (original csla_us85_cfg\config.bin)
Updating base class US85_Launcher_AT->US85_launcherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/US85_MAAWS/ (original csla_us85_cfg\config.bin)
Updating base class US85_Launcher_AT->US85_launcherBase, by z\ace\addons\compat_csla\config.cpp/CfgWeapons/US85_SMAW/ (original csla_us85_cfg\config.bin)
```

after 2.14 we can drop the has include and it will just work
tested and you get a nice `Skipped loading of addon 'ace_compat_csla' as required addon 'CSLA' is not present` in the rpt

if the config changes seem reasonable then I'd like to just get this merged as is and move the todo to a future pr

**Review by PabstMirror — APPROVED** [review]

File: addons/compat_csla/CfgWeapons.hpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        EGVAR(overpressure,damage) = 0.7;
+        EGVAR(overpressure,priority) = 1;
+        EGVAR(overpressure,range) = 10;
+    };
+
+    class US85_SMAW: US85_Launcher_AT {
+        EGVAR(reloadlaunchers,enabled) = 1;
+        EGVAR(overpressure,angle) = 45;
+    };
+
```
Comment (LinkIsGrim) [inline addons/compat_csla/CfgWeapons.hpp:76]:
```suggestion
```

File: addons/compat_csla/CfgWeapons.hpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    class CSLA_LauncherBase;
+    class CSLA_Launcher_AT;
+    class CSLA_MachinegunBase_7_62;
+
+    class CSLA_RPG7: CSLA_Launcher_AT {
+        EGVAR(reloadlaunchers,enabled) = 1;
+    };
+
+    class CSLA_RPG75: CSLA_Launcher_AT {
+
```
Comment (LinkIsGrim) [inline addons/compat_csla/CfgWeapons.hpp:12]: ?

File: addons/compat_csla/CfgWeapons.hpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    };
+
+    class US85_Launcher_AA;
+    class US85_Launcher_AT;
+    class US85_MachinegunBase_5_56;
+    class US85_MachinegunBase_7_62;
+    class US85_weaponBase;
+
+    class US85_M16_base: US85_weaponBase {
+
```
Comment (LinkIsGrim) [inline addons/compat_csla/CfgWeapons.hpp:33]: Missing something here?

File: addons/compat_csla/CfgMagazines.hpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        ace_isBelt = 1;
+    };
+
+    class US85_100Rnd_762x51: US85_20Rnd_762x51 {
+        ace_isBelt = 1;
+    };
+
+    class US85_200Rnd_556x45: US85_30Rnd_556x45 {
+        ace_isBelt = 1;
+    };
```
Comment (LinkIsGrim) [inline addons/compat_csla/CfgMagazines.hpp:23]:
```suggestion
    class CSLA_UK59_50rnd_7_62vz59;
    class CSLA_UK59_50rnd_7_62Sv59: CSLA_UK59_50rnd_7_62vz59 {
        ace_isBelt = 1;
    };

    class US85_Magazine;
    class US85_50Rnd_762x51: US85_Magazine {
        ace_isBelt = 1;
    };

    class US85_20Rnd_762x51;
    class US85_100Rnd_762x51: US85_20Rnd_762x51 {
        ace_isBelt = 1;
    };

    class US85_30Rnd_556x45;
    class US85_200Rnd_556x45: US85_30Rnd_556x45 {
        ace_isBelt = 1;
    };
```

File: addons/compat_csla/CfgWeapons.hpp [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    class US85_M60: US85_MachinegunBase_7_62 {
+        EGVAR(overheating,allowSwapBarrel) = 1;
+        EGVAR(overheating,closedBolt) = 0;
+    };
+
+    class US85_FIM92: US85_Launcher_AA {
+        EGVAR(overpressure,angle) = 45;
+        EGVAR(overpressure,damage) = 0.3;
+        EGVAR(overpressure,range) = 15;
```
Comment (PabstMirror) [inline addons/compat_csla/CfgWeapons.hpp:45]: 15 does seem high for a AA weapon
for ref
```
    class rhs_weap_fim92: launch_O_Titan_F {
        EGVAR(overpressure,range) = 6;
        EGVAR(overpressure,angle) = 40;
        EGVAR(overpressure,damage) = 0.6;
        EGVAR(overpressure,offset) = 1.45;
```

Comment (PabstMirror) [reply addons/compat_csla/CfgWeapons.hpp:45]:
```suggestion
        EGVAR(overpressure,range) = 6;
```

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**